flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/state/PendingSplitsStateSerializer.java

```diff
@@ -306,6 +322,59 @@ private StreamPendingSplitsState deserializeStreamPendingSplitsState(DataInputDe
     // Utilities
     // ------------------------------------------------------------------------------------------
 
+    private void writeSplitFinishedCheckpointIds(
+            Map<String, Long> splitFinishedCheckpointIds, DataOutputSerializer out)
+            throws IOException {
+        final int size = splitFinishedCheckpointIds.size();
+        out.writeInt(size);
+        for (Map.Entry<String, Long> splitFinishedCheckpointId :
+                splitFinishedCheckpointIds.entrySet()) {
+            out.writeUTF(splitFinishedCheckpointId.getKey());
+            out.writeLong(splitFinishedCheckpointId.getValue());
+        }
+    }
+
+    private void writeFinishedSplits(
+            Map<TableId, Set<String>> finishedSplits, DataOutputSerializer out) throws IOException {
+        final int size = finishedSplits.size();
+        out.writeInt(size);
+        for (Map.Entry<TableId, Set<String>> tableSplits : finishedSplits.entrySet()) {
+            out.writeUTF(tableSplits.getKey().toString());
+            out.writeInt(tableSplits.getValue().size());
+            for (String splitId : tableSplits.getValue()) {
+                out.writeUTF(splitId);
+            }
```

Comment:
I'm a little concern about current change which may introduce bigger memory usage of JM side, could we avoid current design and use existed informations?

Reply:
It has been removed.